### ISSUE TITLE
Annotation with generics

### ISSUE THREAD

**Issue by mavimo** [slevomat/coding-standard#297]
Hello!
I try to upgrade to to latest version of this package (version `4.5` from previous `4.1`) but we have some issue with `SlevomatCodingStandard\Sniffs\Namespaces\UnusedUsesSniff`.
We use [phpstan](https://github.com/phpstan/phpstan) via [phropecy](https://github.com/Jan0707/phpstan-prophecy) extension, that allow to use angular brackets to specify a generics for object, eg:

```php
<?php

declare(strict_types=1);

namespace Sample\Tests;

use PHPUnit\Framework\TestCase;
use Prophecy\Prophecy\ObjectProphecy;

class Demo
{
    /**
     * @var int[]
     */
    private $data;

    public function __construct()
    {
        $this->data = [];
    }

    public function add(int $info)
    {
        $this->data[] = $info;
    }

    public function count()
    {
        return count($this->data);
    }
}

final class DemoTest extends TestCase
{
    /**
     * @var ObjectProphecy<Demo>
     */
    private $demo;

    public function setUp(): void
    {
        $this->demo = $this->prophesize(Demo::class);

        $this->demo->count()->willReturn(1);
    }

    public function testDemo(): void
    {
        $this->assertEquals(1, $this->demo->reveal()->count());
    }
}
```
but the specified rules report:
```
FILE: DemoTest.php
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE1
----------------------------------------------------------------------
  8 | ERROR | [x] Type Prophecy\Prophecy\ObjectProphecy is not used
    |       |     in this file.
----------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------

Time: 80ms; Memory: 6Mb
```
the issue seems related to `SlevomatCodingStandard\Helpers\AnnotationHelper` that detect the annotated value `ObjectProphecy<Demo>` instead of two separate annotate objects `ObjectProphect` and `Demo`.

I create a [simple PR](https://github.com/slevomat/coding-standard/pull/296/files) with broken test, let me know if you will accept PR that will allow generics in annotation.

**Comment by kukulich** [comment]
Thank you for your report but is duplicate to https://github.com/slevomat/coding-standard/issues/251

**Comment by mavimo** [comment]
@kukulich sorry I totally miss the #251 :)

**Comment by mavimo** [comment]
@ondrejmirtes I think is done by phpstan extension, I'll investigate.
